alibi_detect/cd/pytorch/preprocess.py

```diff
@@ -7,6 +7,63 @@
                                                    predict_batch_transformer)
 
 
+class _Encoder(nn.Module):
+    def __init__(
+            self,
+            input_layer: Union[nn.Module, int],
+            mlp: Optional[nn.Module] = None,
+            input_dim: Optional[int] = None,
```

Comment:
Additional `input_dim` kwarg (compared to tensorflow version) because unlike keras `Dense` layers, pytorch `Linear` layers need input shape to be given. 

TODO: Explore using `LazyLinear` for consistency with tensorflow implementation...

Reply:
IMO we should stick to `Linear`, since it simplifies `save_model_config`. Compare the [pytorch implementation](https://github.com/ascillitoe/alibi-detect/blob/cf3a9e153d791fe6aa81db1b58f0d46c7269c473/alibi_detect/saving/_pytorch/saving.py#L42-L51) to to the [tensorflow one](https://github.com/SeldonIO/alibi-detect/blob/6df9bd4b0fb6b27a1df41f0cdae76c804dc3cd55/alibi_detect/saving/_tensorflow/saving.py#L55-L69).

Requiring `input_dim` to be passed to `_Encoder` also doesn't significantly complicate the public api, since `_Encoder` is private, and `shape` is already passed to `UAE`.

Reply:
Lazy Linear is also [in active development](https://github.com/pytorch/pytorch/blob/4618371da56c887195e2e1d16dad2b9686302800/torch/nn/modules/lazy.py#L180) and if we did use them we might want to wait until they're stable.